addons/phone_validation/models/phone_blacklist.py

```diff
@@ -40,15 +45,21 @@ def create(self, values):
             done.add(sanitized)
             to_create.append(dict(value, number=sanitized))
 
-        """ To avoid crash during import due to unique email, return the existing records if any """
-        sql = '''SELECT number, id FROM phone_blacklist WHERE number = ANY(%s)'''
-        numbers = [v['number'] for v in to_create]
-        self._cr.execute(sql, (numbers,))
-        bl_entries = dict(self._cr.fetchall())
-        to_create = [v for v in to_create if v['number'] not in bl_entries]
+        # Search for existing phone blacklist entries, even inactive ones (will be activated again)
+        numbers_requested = [values['number'] for values in to_create]
+        existing = self.with_context(active_test=False).search([('number', 'in', numbers_requested)])
 
-        results = super(PhoneBlackList, self).create(to_create)
-        return self.env['phone.blacklist'].browse(bl_entries.values()) | results
+        # Out of existing pb records, activate non-active
+        existing.filtered(lambda pb: not pb.active).write({'active': True})
```

Comment:
Note that technically, we could call create with active=False, and it would activate it instead of bypassing it. But not sure we should support that use case (official API, which is to use add / remove, already handles it).

Reply:
Well, I patched up that case just in case (+test). 

Now it won't reactivate existing numbers if attempted to re-create them with `'active':False`